### PR TITLE
Move Windows Compat pack to NuGet pack task

### PR DESCRIPTION
**Pull request by ViktorHofer**
This is a straight port from pkgproj to csproj and also enables building
the compat pack together with its dependencies.

By default `dotnet pack` doesn't build any of the P2Ps. To build them as well
either use `dotnet build` or `dotnet pack /p:NoBuild=false`.

NuSpec diff: https://www.diffchecker.com/TiSZvOvM

### CODE REVIEW

File: src/libraries/pkg/Microsoft.Windows.Compatibility/Microsoft.Windows.Compatibility.pkgproj [hunk 1]
```diff
@@ -1,98 +0,0 @@
...
-    <BeforeNET6LibraryPackage Include="System.Reflection.Emit" Version="$(SystemReflectionEmitVersion)" />
-    <BeforeNET6LibraryPackage Include="System.Reflection.Emit.ILGeneration" Version="$(SystemReflectionEmitILGenerationVersion)" />
-    <BeforeNET6LibraryPackage Include="System.Reflection.Emit.Lightweight" Version="$(SystemReflectionEmitLightweightVersion)" />
-    <BeforeNET6LibraryPackage Include="System.Security.AccessControl" Version="$(SystemSecurityAccessControlVersion)" />
-    <BeforeNET6LibraryPackage Include="System.Security.Cryptography.Cng" Version="$(SystemSecurityCryptographyCngVersion)" />
-    <BeforeNET6LibraryPackage Include="System.Security.Principal.Windows" Version="$(SystemSecurityPrincipalWindowsVersion)" />
-
-    <!-- External packages -->
-    <ExternalLibraryPackage Include="System.Data.SqlClient" Version="$(SystemDataSqlClientVersion)" />  
-    <ExternalLibraryPackage Include="System.ServiceModel.Primitives;
```
Comment (Anipik) [inline src/libraries/pkg/Microsoft.Windows.Compatibility/Microsoft.Windows.Compatibility.pkgproj:61]: these are missing from the package in the diff for net6.0

Comment (ViktorHofer) [reply src/libraries/pkg/Microsoft.Windows.Compatibility/Microsoft.Windows.Compatibility.pkgproj:61]: Those were missing from the P6 package. In the diff the left side is the P6 package, the right side is the new package.

File: src/libraries/Microsoft.Windows.Compatibility/src/Microsoft.Windows.Compatibility.csproj [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    <ProjectReference Include="$(LibrariesProjectRoot)System.Diagnostics.PerformanceCounter\src\System.Diagnostics.PerformanceCounter.csproj" />
+    <ProjectReference Include="$(LibrariesProjectRoot)System.DirectoryServices.AccountManagement\src\System.DirectoryServices.AccountManagement.csproj" />
+    <ProjectReference Include="$(LibrariesProjectRoot)System.DirectoryServices.Protocols\src\System.DirectoryServices.Protocols.csproj" />
+    <ProjectReference Include="$(LibrariesProjectRoot)System.DirectoryServices\src\System.DirectoryServices.csproj" />
+    <ProjectReference Include="$(LibrariesProjectRoot)System.Drawing.Common\src\System.Drawing.Common.csproj" />
+    <ProjectReference Include="$(LibrariesProjectRoot)System.IO.Packaging\src\System.IO.Packaging.csproj" />
+    <ProjectReference Include="$(LibrariesProjectRoot)System.IO.Ports\src\System.IO.Ports.csproj" />
+    <ProjectReference Include="$(LibrariesProjectRoot)System.Management\src\System.Management.csproj" />
+    <ProjectReference Include="$(LibrariesProjectRoot)System.Reflection.Context\src\System.Reflection.Context.csproj" />
```
Comment (Anipik) [inline src/libraries/Microsoft.Windows.Compatibility/src/Microsoft.Windows.Compatibility.csproj:34]: This should be present for ns2.0

File: src/libraries/pkg/Microsoft.Windows.Compatibility/Microsoft.Windows.Compatibility.pkgproj (hunk 1 above)
Comment (Anipik) [reply src/libraries/pkg/Microsoft.Windows.Compatibility/Microsoft.Windows.Compatibility.pkgproj:61]: okay so thats an intentional diff ?

**Review by Anipik — APPROVED** [review]

File: src/libraries/Microsoft.Windows.Compatibility/src/Microsoft.Windows.Compatibility.csproj (hunk 2 above)
Comment (ViktorHofer) [reply src/libraries/Microsoft.Windows.Compatibility/src/Microsoft.Windows.Compatibility.csproj:34]: As discussed offline, that one is intentionally added as it's compatible with netstandard2.0 and was missing before.

File: src/libraries/Microsoft.Windows.Compatibility/src/Microsoft.Windows.Compatibility.csproj [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    </ProjectReference>
+  </ItemDefinitionGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="$(LibrariesProjectRoot)Microsoft.Win32.Registry.AccessControl\src\Microsoft.Win32.Registry.AccessControl.csproj" />
+    <ProjectReference Include="$(LibrariesProjectRoot)Microsoft.Win32.SystemEvents\src\Microsoft.Win32.SystemEvents.csproj" />
+    <ProjectReference Include="$(LibrariesProjectRoot)System.CodeDom\src\System.CodeDom.csproj" />
+    <ProjectReference Include="$(LibrariesProjectRoot)System.ComponentModel.Composition\src\System.ComponentModel.Composition.csproj" />
+    <ProjectReference Include="$(LibrariesProjectRoot)System.ComponentModel.Composition.Registration\src\System.ComponentModel.Composition.Registration.csproj"
+                      Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'netstandard2.1'))" />
```
Comment (safern) [inline src/libraries/Microsoft.Windows.Compatibility/src/Microsoft.Windows.Compatibility.csproj:26]: Can we move this to it's own item group with the condition on the item group? It makes it easier to read which dependencies apply to which tfms?

**Review by safern — APPROVED** [review]
Other than one comment, looks good. Thanks!

**Comment by ViktorHofer** [comment]
Build: https://dnceng.visualstudio.com/public/_build/results?buildId=1272613&view=results

**Pull request merged**